packages/x-data-grid/src/components/GridScrollArea.tsx

```diff
@@ -164,6 +184,108 @@ function GridScrollAreaContent(props: ScrollAreaProps) {
     });
   });
 
+  return (
+    <GridScrollAreaContent
+      {...props}
+      ref={rootRef}
+      getCanScrollMore={getCanScrollMore}
+      style={style}
+      handleDragOver={handleDragOver}
+    />
+  );
+}
+
+function GridVerticalScrollAreaContent(props: ScrollAreaProps) {
+  const { scrollDirection, scrollPosition } = props;
+  const rootRef = React.useRef<HTMLDivElement>(null);
+  const apiRef = useGridApiContext();
+  const timeout = useTimeout();
+  const rowsMeta = useGridSelector(apiRef, gridRowsMetaSelector);
+
+  const getCanScrollMore = () => {
+    const dimensions = gridDimensionsSelector(apiRef);
+    if (scrollDirection === 'up') {
+      // Only render if the user has not reached yet the top of the list
+      return scrollPosition.current.top > 0;
+    }
+
+    if (scrollDirection === 'down') {
```

Comment:
Noticed an unexpected behavior when scrolling down. It starts scrolling upwards again past a certain threshold:

https://github.com/user-attachments/assets/5e460461-3612-4cd0-a925-0ba5783fd887

Doesn't appear to be a problem with the other directions

Reply:
I couldn't reproduce it in Chrome and Firefox. Did you use a different browser/specific reproduction steps?

Reply:
In Mac OS, it only reproduces for me with this setting under "Scroll bar behavior" section:
![image](https://github.com/user-attachments/assets/25391100-0466-4a9f-a45b-f5b45c559f94)